dali/operators/generic/slice/slice_attr.h

```diff
@@ -119,6 +122,9 @@ class NamedSliceAttr {
         auto axes = axes_;
         if (!axis_names_.empty()) {
           axes = GetDimIndices(shape_layout, axis_names_).to_vector();
+        } else if (axes.empty()) {
```

Comment:
The logic here is broken. It means that if the user types:
`axes=[]` all axes will be used - and in fact, no axis should be used.